### ISSUE TITLE
[api-fix] Sensor statics vs class instance pattern 

### ISSUE THREAD

**Issue by Redth** [xamarin/Essentials#584]
During our API review we debated the problems with having the sensor related classes as static types with static `Start`, `Stop`, `IsMonitoring`, `*Changed` methods/properties/events.

There are a few potential issues here:
 - Another caller may inadvertently call Start or Stop when you are not expecting it
 - Events are global and forgetting to unregister them has a potentially big negative impact

We could instead expose these as class instances which somewhat solves this problem, but not completely.  On Android these sensors are global to the app anyway, and if one caller changes the sensor listener, it changes for ALL listeners.

Let's debate keeping these static (which is a very nice to consume API) vs moving to a type instance pattern.

**Comment by Redth** [comment]
We should add some documentation to remind users that this is a static/global event and they should stop the sensor and unsubscribe from the event when they are done using it!

**Comment by mattleibow** [comment]
I think because of the Android issue, my vote is strongly towards a static class since an inconsistent behaviour is worse than a developer issue with an unsubscribed event.

Also, as this API is geared towards mobile, and mobile only has 1 sensor of each type, the complexity of multiple sensor is probably not worth it.

**Comment by Mrnikbobjeff** [comment]
My biggest issue with static classes is that I usually have to wrap Essentials classes behind a proxy. This way I can still use it with DI and have it testable. This is also my biggest issue with Xamarin.Essentials overall. For every sensor we used we created such a proxy object. To alleviate your concerns one could use a factory pattern coupled with a private constructor for people to always get the same object without it being a static class. I would love if there could be some sort of discussion about making Essentials better to work with from a DI perspective :)
